src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final String COMMA_SEPARATOR = ",";
+    private static final int OPERATION_TYPE_POSITION = 0;
+    private static final int AMOUNT_POSITION = 1;
+    private static final OperationType SUPPLY = OperationType.SUPPLY;
+    private static final OperationType BUY = OperationType.BUY;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String[] data = getDataFromFile(fromFileName);
+        for (String dataObject : data) {
+            String[] splittedDataString = dataObject.split(COMMA_SEPARATOR);
+            if (splittedDataString[OPERATION_TYPE_POSITION]
+                    .equals(SUPPLY.toString().toLowerCase())) {
+                supplyAmount += Integer.parseInt(splittedDataString[AMOUNT_POSITION]);
+            }
+            if (splittedDataString[OPERATION_TYPE_POSITION]
+                    .equals(BUY.toString().toLowerCase())) {
+                buyAmount += Integer.parseInt(splittedDataString[AMOUNT_POSITION]);
+            }
+        }
```

Comment:
Please move this part of logic to createReport method to your public method as short and simple as possible:
String[] data = getDataFromFile(fromFileName);
writeDataToFile(toFileName, createReport(data));